jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2818,11 +2818,62 @@ public static MethodHandle dropArguments(MethodHandle originalHandle, int locati
 	 * @param originalHandle the original method handle to be transformed
 	 * @param skippedArgumentCount the number of argument to be skipped from the original method handle
 	 * @param valueTypes a List of the argument types to be inserted
-	 * @param location the location of the first argument to be removed
+	 * @param location the (zero-indexed) location of the first argument to be removed
 	 * @return a MethodHandle representing a transformed handle as described above
 	 */
-	public static MethodHandle dropArgumentsToMatch(MethodHandle originalHandle, int skippedArgumentCount, List<Class<?>> valueTypes, int location) {
-		throw new UnsupportedOperationException("The method has not yet been implemented for now"); //$NON-NLS-1$
+	public static final  MethodHandle dropArgumentsToMatch(MethodHandle originalHandle, int skippedArgumentCount, List<Class<?>> valueTypes, int location) {
+		/* implicit null checks */
+		MethodType originalType = originalHandle.type;
+		Class<?>[] valueTypesCopy = new Class<?>[valueTypes.size()];
+
+		/* check if indexing is in range*/
+		if ((0 > skippedArgumentCount) ||
+			(skippedArgumentCount > originalType.parameterCount())) {
+			throw new IllegalArgumentException("Invalid parameters: skippedArgumentCount"); //$NON-NLS-1$
+		}
+		if ((0 > location) ||
+			(valueTypes.size() < location) ||
```

Comment:
if location is zero indexed then wouldn't "valueTypes.size() <= location" also be out of range?

Reply:
I discussed with Jack and the second check ensures that location never overflows